Userland/Libraries/LibPDF/Filter.cpp

```diff
@@ -114,10 +116,19 @@ PDFErrorOr<ByteBuffer> Filter::decode_ascii85(ReadonlyBytes bytes)
 
         u32 number = 0;
 
-        auto const to_write = byte_index + 5 >= bytes.size() ? bytes.size() - byte_index : 5;
+        auto to_write = byte_index + 5 >= bytes.size() ? bytes.size() - byte_index : 5;
+
+        Optional<u32> end_of_data_index {};
 
         for (int i = 0; i < 5; i++) {
-            auto byte = byte_index >= bytes.size() ? 'u' : bytes[byte_index++];
+            if (!end_of_data_index.has_value() && bytes[byte_index] == '~') {
```

Comment:
I suppose just checking for `~` instead of `~>` is enough since `>` is the only thing that can follow `>`, but it's a bit subtle. Maybe add a comment along the lines of `// '~>' is the end-of-data marker. '~' is only valid if followed by '>' so checking for '~' is sufficient.`?